sap-abap/services/ec2/zcl_aws1_ec2_actions.clas.testclasses.abap

```diff
@@ -0,0 +1,521 @@
+    DATA(lo_result) = ao_ec2_actions->describe_key_pairs( ).
+
+    DATA lv_found TYPE abap_bool VALUE abap_false.
+    LOOP AT lo_result->get_keypairs( ) INTO DATA(lo_key_pair).
+      IF lo_key_pair->get_keyname( ) = cv_key_name.
+        lv_found = abap_true.
+      ENDIF.
+    ENDLOOP.
+
+    cl_abap_unit_assert=>assert_true(
+      act = lv_found
+      msg = |Key Pair { cv_key_name } should have been included in key pair list|
+    ).
+    ao_ec2->deletekeypair( iv_keyname =  cv_key_name ).
+  ENDMETHOD.
+  METHOD describe_regions.
+    DATA(lo_result) = ao_ec2_actions->describe_regions( ).
+    cl_abap_unit_assert=>assert_not_initial(
+          act = lo_result->get_regions( )
+          msg = |Failed to retrieve list of regions|
+        ).
+  ENDMETHOD.
+  METHOD describe_availability_zones.
+    DATA(lo_result) = ao_ec2_actions->describe_availability_zones( ).
+    cl_abap_unit_assert=>assert_not_initial(
+          act = lo_result->get_availabilityzones( )
+          msg = |Failed to retrieve list of availability zones|
+        ).
+  ENDMETHOD.
+  METHOD create_security_group.
+    CONSTANTS cv_security_group_name TYPE /aws1/ec2string VALUE 'code-example-create-security-group'.
+    DATA(lv_vpc_id) = ao_ec2->createvpc( iv_cidrblock  = '10.1.0.0/16' )->get_vpc( )->get_vpcid( ).
+    DATA(lo_create_result) = ao_ec2_actions->create_security_group( iv_security_group_name =  cv_security_group_name iv_vpc_id = lv_vpc_id ).
+    DATA(lo_describe_result) = ao_ec2->describesecuritygroups(
+      it_groupids = VALUE /aws1/cl_ec2groupidstrlist_w=>tt_groupidstringlist(
+                      ( NEW /aws1/cl_ec2groupidstrlist_w( lo_create_result->get_groupid( ) ) )
+                    )
+    ).
+
+    DATA lv_found TYPE abap_bool VALUE abap_false.
+    LOOP AT lo_describe_result->get_securitygroups( ) INTO DATA(lo_security_group).
+      IF lo_security_group->get_groupname( ) = cv_security_group_name.
+        lv_found = abap_true.
+      ENDIF.
+    ENDLOOP.
+
+    cl_abap_unit_assert=>assert_true(
+      act = lv_found
+      msg = |Failed to create security group { cv_security_group_name }|
+    ).
+
+    ao_ec2->deletesecuritygroup( iv_groupid = lo_create_result->get_groupid( ) ).
+    ao_ec2->deletevpc( iv_vpcid = lv_vpc_id ).
+  ENDMETHOD.
+  METHOD delete_security_group.
+    CONSTANTS cv_security_group_name TYPE /aws1/ec2string VALUE 'code-example-delete-security-group'.
+    DATA(lv_vpc_id) = ao_ec2->createvpc( iv_cidrblock  = '10.2.0.0/16' )->get_vpc( )->get_vpcid( ).
+    DATA(lo_create_result) = ao_ec2->createsecuritygroup(
+        iv_groupname =  cv_security_group_name
+        iv_description = |security group for delete_security_group test|
+        iv_vpcid = lv_vpc_id
+    ).
+    ao_ec2_actions->delete_security_group( iv_security_group_id = lo_create_result->get_groupid( ) ).
+    DATA(lo_describe_result) = ao_ec2->describesecuritygroups(
+      it_filters = VALUE /aws1/cl_ec2filter=>tt_filterlist(
```

Comment:
This block formatting feels weird? I don't think I've seen it elsewhere, but I haven't seen a list initializer yet